### PR TITLE
bls12: changing serialization of Fp to big-endian format

### PR DESCRIPTION
**Pull request by armfazh**
These changes are needed to accomplish #257 .

### CODE REVIEW

File: ecc/bls12381/gt.go [hunk 1]
```diff
@@ -8,14 +8,16 @@ const GtSize = ff.URootSize
...
-func (z *Gt) Set(x *Gt)                   { z.i.Set(&x.i) }
-func (z *Gt) SetBytes(b []byte) error     { return z.i.SetBytes(b) }
-func (z Gt) Bytes() []byte                { return z.i.Bytes() }
-func (z *Gt) SetIdentity()                { z.i.SetIdentity() }
-func (z Gt) IsEqual(x *Gt) bool           { return z.i.IsEqual(&x.i) == 1 }
-func (z Gt) IsIdentity() bool             { i := &Gt{}; i.SetIdentity(); return z.IsEqual(i) }
-func (z *Gt) Mul(x, y *Gt)                { z.i.Mul(&x.i, &y.i) }
-func (z *Gt) Sqr(x *Gt)                   { z.i.Sqr(&x.i) }
-func (z *Gt) Inv(x *Gt)                   { z.i.Inv(&x.i) }
+func (z Gt) String() string           { return z.i.String() }
```
Comment (wbl) [inline ecc/bls12381/gt.go:11]: I'm not sure why these changed. And now that I look at then I'm not sure why we pass-by-value sometimes.

**Review by wbl — APPROVED** [review]

**Comment by wbl** [comment]
Would be nice to see test vectors confirming interop in tests.

File: ecc/bls12381/gt.go (hunk 1 above)
Comment (armfazh) [reply ecc/bls12381/gt.go:11]: As it is passed by value, this ensures that the method will not modify the value (by-accident).

**Comment by armfazh** [comment]
> Would be nice to see test vectors confirming interop in tests.

Test vectors were added in #273 .

**Pull request merged**